applications/luci-app-apinger/htdocs/luci-static/resources/view/apinger/targets.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+'require view';
+'require form';
+'require uci';
+
+var a_ifaces;
+var a_down;
+var a_delay;
+var a_loss;
+
+return view.extend({
+	load: function() {
+		return Promise.all([
+			uci.load('apinger'),
+		])
+	},
+
+	renderTargets: function(m) {
+		var target, _interface, address, probe_interval;
+		var avg_delay_samples, avg_loss_samples, avg_loss_delay_samples;
+		var alarm_down, alarm_delay, alarm_loss;
+
+		target = m.section(form.GridSection, 'target');
+		target.anonymous = false;
+		target.addremove = true;
+		target.addbtntitle = _('Add Target');
+
+		_interface = target.option(form.Value, 'interface', _('Interface'));
+		for (var i = 0; i < a_ifaces.length; i++) {
+			_interface.value(a_ifaces[i]['.name']);
+		}
+
+		address = target.option(form.Value, 'address', _('Address'));
+		address.datatype = 'ip4addr';
+
+		probe_interval = target.option(form.Value, 'probe_interval', _('Ping Interval'));
+		probe_interval.datatype = 'integer';
+
+		avg_delay_samples = target.option(form.Value, 'avg_delay_samples', _('Average Delay'));
+		avg_delay_samples.datatype = 'integer';
+
+		avg_loss_samples = target.option(form.Value, 'avg_loss_samples', _('Average Loss'));
+		avg_loss_samples.datatype = 'integer';
+
+		avg_loss_delay_samples = target.option(form.Value, 'avg_loss_delay_samples', _('Average Loss/Delay'));
+		avg_loss_delay_samples.datatype = 'integer';
+
+		alarm_down = target.option(form.Value, 'alarm_down', _('Down Alarm'));
+		for (var i = 0; i < a_down.length; i++) {
+			alarm_down.value(a_down[i]['.name']);
+		}
+
+		alarm_delay = target.option(form.Value, 'alarm_delay', _('Delay Alarm'));
+		for (var i = 0; i < a_delay.length; i++) {
+			alarm_delay.value(a_delay[i]['.name']);
+		}
+
+		alarm_loss = target.option(form.Value, 'alarm_loss', _('Loss Alarm'));
+		for (var i = 0; i < a_loss.length; i++) {
+			alarm_loss.value(a_loss[i]['.name']);
+		}
+
+		return m;
+	},
+
+	render: function(data) {
+		var m;
+
+		a_ifaces = uci.sections('apinger', 'interface');
+		a_down = uci.sections('apinger', 'alarm_down');
+		a_delay = uci.sections('apinger', 'alarm_delay');
+		a_loss = uci.sections('apinger', 'alarm_loss');
+
+		m = new form.Map('apinger', _('Apinger - Targets'),
+			_('Interface: Interface to use to track target') + '<br />' +
+			_('Address: Target address to be tracked') + '<br />' +
+			_('Ping Interval: How often the probe should be sent') + '<br />' +
+			_('Average Delay: How many replies should be used to compute average delay') + '<br />' +
+			_('Average Loss: How many probes should be used to compute average loss') + '<br />' +
+			_('Average Delay and Loss: The delay (in samples) after which loss is computed, without this delays larger than interval would be treated as loss') +
+			'<br />');
+		m = this.renderTargets(m);
```

Comment:
Why do you have an extra function in all views? I would prefer the content of the called function to be inserted here. Instead of calling this. This would also allow you to keep the variables local.

Reply:
Updated